clang/lib/DPCT/DPCT.cpp

```diff
@@ -846,17 +861,20 @@ int runDPCT(int argc, const char **argv) {
     }
   }
 
-  if ((BuildScript == BuildScriptKind::BS_Cmake ||
-       BuildScript == BuildScriptKind::BS_Python) &&
-      !OptParser->getSourcePathList().empty()) {
-    ShowStatus(MigrateBuildScriptIncorrectUse);
-    dpctExit(MigrateBuildScriptIncorrectUse);
-  }
-  if ((BuildScript == BuildScriptKind::BS_Cmake ||
-       BuildScript == BuildScriptKind::BS_Python) &&
-      MigrateBuildScriptOnly) {
-    ShowStatus(MigrateBuildScriptAndMigrateBuildScriptOnlyBothUse);
-    dpctExit(MigrateBuildScriptAndMigrateBuildScriptOnlyBothUse);
+  DpctGlobalInfo::setBuildScript(BuildScript.getBits());
+  bool BuildScriptsSpecified = DpctGlobalInfo::migrateCMakeScripts() ||
+                               DpctGlobalInfo::migratePythonScripts();
+  if (MigrateBuildScriptOnly) {
+    if (!BuildScriptsSpecified) {
+      llvm::errs() << getBuildScriptNotSpecifiedWarning();
+      DpctGlobalInfo::setBuildScript(1); // enable cmake script migration
```

Comment:
```suggestion
      DpctGlobalInfo::setBuildScript(1); // Please update magic number to meaning macro.
```

Reply:
Added the logic to calculate bits value for CMake script selection (**1 << BS_CMake**)